src/API.js

```diff
@@ -1,38 +1,60 @@
-const baseUrl = "https://auth.api.dev.mindtastic.lol";
+const baseUrl = "https://auth.api.%CLUSTER%.mindtastic.lol";
+const proxyUrl = "/%CLUSTER%";
+const endpoints = {
+    regInit: "/self-service/registration/%FLOW%",
+    regSubmit: "/self-service/registration?flow=%FLOWID%",
+    loginInit: "/self-service/login/%FLOW%",
+    loginSubmit: "/self-service/login?flow=%FLOWID%",
+    logoutInit: "/self-service/logout/%FLOW%",
+    logoutSubmit: "/self-service/logout?token=%LOGOUTTOKEN%",
+    whoami: "/sessions/whoami"
+}
 
 export default {
 
+    endpoints: endpoints,
     baseUrl: baseUrl,
 
-    initRegistration: async () => {
-        const response = await _fetch('/self-service/registration/api');
-        const r = await response.json();
-        return r.ui.action;
+    makeRequest: async (request) => {
+        var response = await fetch(request.proxyUrl, request);
+
+        response = await response.json()
+            .then(json => ({
+                headers: response.headers,
+                status: response.status,
+                json
+            }))
+            .catch((err) => console.log(err));
+
+        return response;
```

Comment:
```suggestion
      let json = {};

      try {
          const response = await fetch(request.proxyUrl, request);
          json = await response.json();
      } catch (err) {
        console.err(err);
      }

      return {
          headers: response.headers,
          status: response.status,
          json,
        };
```

- Always use `let` or `const` over `var`, unless there is a very good reason (You know if it's the case).
- I don't like mixing up the `async/await` and the promise syntax. I understand if the promise approach is preferred, but we should stay consistent :)

- This is no proper error handling, but as long as we don't provide this as a library, it should be fine. We should think about *rethrowing* the error.